genie/bed.py

```diff
@@ -480,19 +458,40 @@ def _process(
         # Must be .astype(bool) because `1, 0 in [True, False]`
         bed[4] = bed[4].astype(bool)
 
-        exon_gtf_path, gene_gtf_path = \
-            create_gtf(process_functions.SCRIPT_DIR)
+        exon_gtf_path, gene_gtf_path = create_gtf(process_functions.SCRIPT_DIR)
+        LOGGER.info("REMAPPING %s" % seq_assay_id)
+        # bedname = seq_assay_id + ".bed"
+        bed.columns = ["Chromosome", "Start_Position", "End_Position",
+                       "Hugo_Symbol", "includeInPanel", "clinicalReported"]
+        # Validate gene symbols
+        # Gene symbols can be split by ; and _ and : and .
+        bed['Hugo_Symbol'] = [
+            i.split(";")[0].split("_")[0].split(":")[0].split(".")[0]
+            for i in bed['Hugo_Symbol']]
+        # Replace all chr with blank
+        bed['Chromosome'] = [
+            str(i).replace("chr", "") for i in bed['Chromosome']]
+        # Change all start and end to int
+        bed['Start_Position'] = bed['Start_Position'].apply(int)
+        bed['End_Position'] = bed['End_Position'].apply(int)
 
-        bed, temp_bed_path = self.createdBEDandGenePanel(
-            bed, seq_assay_id, genePanelPath, parentId, exon_gtf_path,
-            createGenePanel=createPanel)
-        genie_combined_path = \
-            add_feature_type(temp_bed_path, exon_gtf_path, gene_gtf_path)
+        genePosition = self.syn.tableQuery('SELECT * FROM syn11806563')
+        genePositionDf = genePosition.asDataFrame()
+        bed['ID'] = bed['Hugo_Symbol']
+        bed = bed.apply(lambda x: validate_symbol(x, genePositionDf,
+                                                  return_mappeddf=True), axis=1)
+        bed['SEQ_ASSAY_ID'] = seq_assay_id
+        temp_bed_path = os.path.join(process_functions.SCRIPT_DIR, "temp.bed")
+        bed.to_csv(temp_bed_path, sep="\t", index=False, header=None)
+        self.create_gene_panel(temp_bed_path, seq_assay_id,
```

Comment:
this should return something and not do a `syn.store` inside - you already noted that in 489 you are duplicating work.

Reply:
Resolve with #203